bokeh/models/widgets/inputs.py

```diff
@@ -43,6 +43,10 @@ class TextInput(InputWidget):
     A callback to run in the browser whenever the user unfocuses the TextInput
     widget by hitting Enter or clicking outside of the text box area.
     """)
+    
+    placeholder = String(default="", help="""
+    Placeholder for empty input field
+    """)
```

Comment:
@busino it occurs to me, does placeholder need to be a single character? If so we can add a new class Character(String) property type that validates len(self.value)==1 easily enough.